clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -1864,16 +1864,20 @@ static void printArguments(ASTContext &Ctx, raw_ostream &ArgOS,
                            const PrintingPolicy &P);
 
 static void printArgument(ASTContext &Ctx, raw_ostream &ArgOS,
-                          TemplateArgument Arg, const PrintingPolicy &P) {
+                          TemplateArgument Arg, const PrintingPolicy &P,
+                          bool FirstArg) {
   switch (Arg.getKind()) {
   case TemplateArgument::ArgKind::Pack: {
+    if (Arg.pack_size() && !FirstArg)
+      ArgOS << ", ";
```

Comment:
What gets printed when size is 0 in the print statement below?

Reply:
Nothing. The function returns without printing anything since it won't enter the loop. I can nest this under an if statement if you'd like.

Reply:
No, I think it is fine.   I was wondering if we have a case of missing/extraneous comma if the pack was empty but there was a deducible parameter after it.

Reply:
Could you provide a test please? I'm not sure how to write this. I can verify/add it.

Reply:
I was thinking of something like this:
```
bash > c++filt _Z1fIiJEiEvT_T1_DpT0_
void f<int, , int>(int, int)
bash > cat foo.cpp
template <typename X, typename ...P, typename Y>
void f(X, Y, P...);

void foo() {
  f(1, 4);
}
```

Reply:
Could you please confirm if this is applicable for class templates. I was trying to find the syntax for this and came across this in cppreference - "In a primary class template, the template parameter pack must be the final parameter in the template parameter list. In a function template, the template parameter pack may appear earlier in the list provided that all following parameters can be deduced from the function arguments, or have default arguments"